WowApiWrapper.lua

```diff
@@ -0,0 +1,82 @@
+WowApiWrapper = {}
+WowApiWrapper.__index = WowApiWrapper
+
+local instance;
+
+function WowApiWrapper:create()
+    -- We don't want to make more than one instance of the API wrapper
+    -- as there's no need.
+
+    -- Instead, store the one we created locally and if someone asks for a new
+    -- one, just give them the one we already have.
+    if not instance then
```

Comment:
Uses a really simplified singleton style setup to make sure we only ever go through the work of connecting to the global WoW API one time.